test/contract/auditing/auditing_test.go

```diff
@@ -0,0 +1,144 @@
+package auditing
+
+import (
+	"context"
+	_ "embed"
+	"log"
+	"testing"
+	"time"
+
+	"github.com/mongodb/mongodb-atlas-kubernetes/v2/internal/translation/auditing"
+	"github.com/mongodb/mongodb-atlas-kubernetes/v2/test/helper/control"
+	"github.com/mongodb/mongodb-atlas-kubernetes/v2/test/helper/launcher"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/mongodb/mongodb-atlas-kubernetes/v2/test/contract"
+)
+
+//go:embed test.yml
+var testYml string
+
+const (
+	testVersion = "2.1.0"
+)
+
+func TestMain(m *testing.M) {
+	if !control.Enabled("AKO_CONTRACT_TEST") {
+		log.Printf("Skipping contract test as AKO_CONTRACT_TEST is unset")
+		return
+	}
+	l := launcher.NewFromEnv(testVersion)
+	l.Launch(
+		testYml,
+		launcher.WaitReady("atlasprojects/my-project", 30*time.Second))
+	if !control.Enabled("SKIP_CLEANUP") { // allow to reuse Atlas resources for local tests
+		defer l.Cleanup()
+	}
+	m.Run()
```

Comment:
if you use `testing.M` you must `os.Exit` with some exit code otherwise tests will never yield exit code 1, even if tests fail. As per godoc:
```
$ go doc testing
...
TestMain runs in the main goroutine and can do whatever
setup and teardown is necessary around a call to m.Run. m.Run will return an
exit code that may be passed to os.Exit. If TestMain returns, the test wrapper
will pass the result of m.Run to os.Exit itself.
...
$ go doc testing.M.Run
package testing // import "testing"

func (m *M) Run() (code int)
    Run runs the tests. It returns an exit code to pass to os.Exit.
```
so you at least need:
```suggestion
	os.Exit(m.Run())
```

Reply:
Coming next